src/main/java/com/bootcamp/demo/service/ScooterServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+    private static final ResponseEntity<Object> SUCCESS_RESPONSE = new ResponseEntity<>(HttpStatus.OK);
+    private static final ResponseEntity<Object> FAILURE_RESPONSE = new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+
+    private ScooterServiceImpl(Firestore db) {
+        this.db = db;
+    }
+
+    @Override
+    public ResponseEntity<Scooter> findScooterById(String id) {
+        LOGGER.info("FIND SCOOTER BY ID - service function invoked");
+        return null;
+    }
+
+    @Override
+    public Set<Scooter> findAllScooters() {
+        LOGGER.info("FIND ALL SCOOTER - service function invoked");
+
+        try {
+            ApiFuture<QuerySnapshot> querySnapshotApiFuture  = db.collection(COLLECTION_SCOOTERS_PATH).get();
+
+            Iterable<QueryDocumentSnapshot> scootersDocuments = querySnapshotApiFuture.get().getDocuments();
+
+            LOGGER.info("Found scooters successfully");
+
+            return StreamSupport
+                    .stream(scootersDocuments.spliterator(), false)
+                    .map(ScooterAssembler::documentToModel)
+                    .collect(Collectors.toSet());
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error(e.getMessage());
+            throw new ServiceException();
+        }
+
+    }
+
+    @Override
+    public void createScooter(final Scooter scooter) {
+        LOGGER.info("CREATE SCOOTER - service function invoked");
+
+        try {
+            ApiFuture<WriteResult> collectionsApiFuture = db.collection(COLLECTION_SCOOTERS_PATH)
+                    .document(scooter.getSerialNumber())
+                    .set(scooter);
+
+            String update_time = collectionsApiFuture.get().getUpdateTime().toDate().toString();
+
+            LOGGER.info("SCOOTER create successfully. Update_time: " + update_time);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error(e.getMessage());
+            throw new ServiceException();
+        }
+    }
+
+    private boolean approximateLocation(Location userLocation, Location scooterLocation) {
+        if (scooterLocation.getLongitude() <= userLocation.getLongitude() + 0.01
+            && (scooterLocation.getLongitude() > userLocation.getLongitude() - 0.01 )) {
+            if ((userLocation.getLatitude() + 0.01 >= scooterLocation.getLatitude())
+                && (userLocation.getLatitude() - 0.01 < scooterLocation.getLatitude())) {
+                return true;
+            }
+        }
+        System.out.println("The scooter is not nearby");
+        return false;
+    }
+
+    @Override
+    public Set<Scooter> getAvailableScooters(Location userLocation) {
+        LOGGER.info("GET AVAILABLE SCOOTERS - service function invoked");
+        Set<Scooter> allScooters = findAllScooters();
+        Set<Scooter> availableScooters = new HashSet<>();
+        for (Scooter scooter : allScooters) {
```

Comment:
You could try rewriting the for using stream functionalities. If you would like, I could reference some articles/urls where you would find details about this. 
Optional, but I thought it might be a nice exercise since perhaps you have not used that very much.